eclipse-collections/src/main/java/org/eclipse/collections/impl/multimap/AbstractSynchronizedMultimap.java

```diff
@@ -0,0 +1,399 @@
+    {
+        synchronized (this.lock)
+        {
+            return this.delegate.multiValuesView();
+        }
+    }
+
+    @Override
+    public RichIterable<V> valuesView()
+    {
+        synchronized (this.lock)
+        {
+            return this.delegate.valuesView();
+        }
+    }
+
+    @Override
+    public RichIterable<Pair<K, RichIterable<V>>> keyMultiValuePairsView()
+    {
+        synchronized (this.lock)
+        {
+            return this.delegate.keyMultiValuePairsView();
+        }
+    }
+
+    @Override
+    public RichIterable<Pair<K, V>> keyValuePairsView()
+    {
+        synchronized (this.lock)
+        {
+            return this.delegate.keyValuePairsView();
+        }
+    }
+
+    @Override
+    public MutableMap<K, RichIterable<V>> toMap()
+    {
+        synchronized (this.lock)
+        {
+            return this.delegate.toMap();
+        }
+    }
+
+    @Override
+    public <R extends Collection<V>> MutableMap<K, R> toMap(Function0<R> collectionFactory)
+    {
+        synchronized (this.lock)
+        {
+            return this.delegate.toMap(collectionFactory);
+        }
+    }
+
+    @Override
+    public <R extends MutableMultimap<K, V>> R selectKeysValues(Predicate2<? super K, ? super V> predicate, R target)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().selectKeysValues(predicate, target);
+        }
+    }
+
+    @Override
+    public <R extends MutableMultimap<K, V>> R rejectKeysValues(Predicate2<? super K, ? super V> predicate, R target)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().rejectKeysValues(predicate, target);
+        }
+    }
+
+    @Override
+    public <R extends MutableMultimap<K, V>> R selectKeysMultiValues(Predicate2<? super K, ? super Iterable<V>> predicate, R target)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().selectKeysMultiValues(predicate, target);
+        }
+    }
+
+    @Override
+    public <R extends MutableMultimap<K, V>> R rejectKeysMultiValues(Predicate2<? super K, ? super Iterable<V>> predicate, R target)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().rejectKeysMultiValues(predicate, target);
+        }
+    }
+
+    @Override
+    public <K2, V2, R extends MutableMultimap<K2, V2>> R collectKeysValues(Function2<? super K, ? super V, Pair<K2, V2>> function, R target)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().collectKeysValues(function, target);
+        }
+    }
+
+    @Override
+    public <V2, R extends MutableMultimap<K, V2>> R collectValues(Function<? super V, ? extends V2> function, R target)
+    {
+        synchronized (this.lock)
+        {
+            return this.getDelegate().collectValues(function, target);
+        }
+    }
+
+    protected MutableMultimap<K, V> getDelegate()
```

Comment:
Please move this to the top. 
Above `getLock()` around line 51 will be an appropriate location. This is to maintain the order of methods.